polar-core/src/vm.rs

```diff
@@ -1432,14 +1481,30 @@ impl PolarVirtualMachine {
     /// consists of unifying the rule head with the arguments, then
     /// querying for each body clause.
     fn query(&mut self, term: &Term) -> Result<QueryEvent> {
-        // Don't log if it's just a single element AND like lots of rule bodies tend to be.
+        // print INFO event for the first call `query` with a rule
+        // (`Call(predicate)`) in the policy execution, ignoring subsequent
+        // calls nested in the execution.
+        //
+        // print TRACE (a superset of INFO) event for all the subsequent calls
+        // We filter out "single element AND" which many rule bodies
+        // take the form of to instead log only their inner operations for
+        // readibility|brevity reasons.
```

Comment:
Took a stab at cleaning up these comments a bit:

```suggestion
        // - Print INFO event for the top-level query for a rule.
        // - Print TRACE (a superset of INFO) event for all subsequent calls.
        // - We filter out single-element ANDs, which many rule bodies take the form of, to instead
        //   log only their inner operations for readibility|brevity reasons.
```